framework/src/engine/consensus/consensus.ts

```diff
@@ -252,7 +252,7 @@ export class Consensus {
 		}
 
 		if (!Buffer.isBuffer(data)) {
-			const errorMessage = 'Received invalid post block data';
+			const errorMessage = 'Received invalid post block data. Applying penalty on peer';
```

Comment:
As confirmed by @mitsuaki-u, the more correct English phrase would be

`Applying a penalty to the peer.`

Also, don't we have a standard that each message should end with a dot?